### PR TITLE
Articles update refactoring

### PR DESCRIPTION
**Pull request by lightalloy**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
 Made changes to `Articles::Updater` in `ArticlesController#update` (not only in `Api::V0::ArticlesController#update`) to avoid duplicating logic

## Related Tickets & Documents
#3811 

## Added tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [x] This change does not need to be communicated, and this is why not: no actual changes in the behavior.

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -240,29 +232,13 @@ def set_article
...
 
-  def article_params
-    params[:article][:published] = true if params[:submit_button] == "PUBLISH"
-    modified_params = policy(Article).permitted_attributes
-    modified_params << :user_id if org_admin_user_change_privilege
-    modified_params << :comment_template if current_user.has_role?(:admin)
-    params.require(:article).permit(modified_params)
-  end
-
   # TODO: refactor all of this update logic into the Articles::Updater possibly,
```
Comment (lightalloy) [inline app/controllers/articles_controller.rb:235]: I still kept some logic here, so kept the comment. There is room for future improvements.

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -127,16 +127,8 @@ def update
     authorize @article
     @user = @article.user || current_user
 
-    not_found if @article.user_id != @user.id && !@user.has_role?(:super_admin)
```
Comment (lightalloy) [inline app/controllers/articles_controller.rb:130]: Other articles should be filtered out by `authorize`, so no need to check one more time.

Comment (citizen428) [reply app/controllers/articles_controller.rb:130]: Not part of this PR, but this got me wondering if in the long run, we should actually use Pundit's `policy_scope` and get rid of the `before_action` that sets `@article`.

File: app/services/articles/attributes.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    attr_reader :attributes, :article_user
+
+    def initialize(attributes, article_user)
+      @attributes = attributes
+      @article_user = article_user
+    end
+
+    def for_update(update_edited_at: false)
+      hash = {}
```
Comment (citizen428) [inline app/services/articles/attributes.rb:15]: I may be misreading this but I think the first 4 lines basically boil down to something like this (`hash` contains only the key-value-pairs from `attributes` for the keys listed in `ATTRIBUTES`):

```ruby
hash = attributes.slice(*ATTRIBUTES)
```

File: app/services/articles/attributes.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      hash[:edited_at] = Time.current if update_edited_at
+      hash
+    end
+
+    private
+
+    def collection
+      if attributes[:series].present?
+        Collection.find_series(attributes[:series], article_user)
+      elsif attributes[:series] == "" # reset collection?
```
Comment (citizen428) [inline app/services/articles/attributes.rb:31]: And if that doesn't match already returns `nil`, so unless I'm missing something this `elsif` doesn't really provide any functionality right now.

Comment (lightalloy) [reply app/services/articles/attributes.rb]: You're right, in all other cases it'll be `nil` anyway; fixed.

**Review by rhymes — APPROVED** [review]
LGTM!

🎉

**Review by Ridhwana — COMMENTED** [review]
LGTM!

**Pull request merged**